File: app/src/main/java/com/example/firebasewithkotlin/Adapter/UserAdapter.java
package com.example.firebasewithkotlin.Adapter;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.firebasewithkotlin.DetailedActivity;
import com.example.firebasewithkotlin.Model.User;
import com.example.firebasewithkotlin.R;
import com.example.firebasewithkotlin.network.Utils;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.android.gms.tasks.Task;

import java.util.List;

public class UserAdapter extends RecyclerView.Adapter<UserAdapter.UserViewHolder> {
    Context context;
    List<User> userList;

    public UserAdapter(Context context, List<User> userList) {
        this.context = context;
        this.userList = userList;
    }

    @NonNull
    @Override
    public UserViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.single_cardview, parent, false);
        return new UserViewHolder(view);

    }

    @Override
    public void onBindViewHolder(@NonNull UserViewHolder holder, int position) {
        holder.textView.setText(userList.get(position).getName() + " " + userList.get(position).getNumber());
        holder.textView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent=new Intent(context, DetailedActivity.class);
                intent.putExtra("uid",userList.get(position).getUid());
                context.startActivity(intent);
            }
        });
        holder.textView.setOnLongClickListener(new View.OnLongClickListener() {
            @Override
            public boolean onLongClick(View v) {
                String uid=userList.get(position).getUid();
                Task<Void> task= Utils.removeUser(uid);
                task.addOnSuccessListener(new OnSuccessListener<Void>() {
                    @Override
                    public void onSuccess(Void aVoid) {
                        Toast.makeText(context, "Removed Successfully", Toast.LENGTH_SHORT).show();
                    }
                });
                return true;
            }
        });


    }

    @Override
    public int getItemCount() {
        return userList.size();
    }

    class UserViewHolder extends RecyclerView.ViewHolder {
        TextView textView;

        public UserViewHolder(@NonNull View itemView) {
            super(itemView);
            textView = itemView.findViewById(R.id.textView);
        }
    }
}

File: app/src/main/java/com/example/firebasewithkotlin/Adapters/UserAdapter.kt
package com.example.firebasewithkotlin.Adapters

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.example.firebasewithkotlin.Extensions.log
import com.example.firebasewithkotlin.Extensions.showToast
import com.example.firebasewithkotlin.R
import com.example.firebasewithkotlin.model.User
import com.example.firebasewithkotlin.network.removeUser
import com.google.android.gms.tasks.Task
import kotlinx.android.synthetic.main.single_cardview.view.*

class UserAdapter(var list: List<User>, var context: Context) :
    RecyclerView.Adapter<UserAdapter.UserViewHolder>() {


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UserViewHolder {
        var view: View =
            LayoutInflater.from(context).inflate(R.layout.single_cardview, parent, false)
        return UserViewHolder(view)
    }

    override fun getItemCount(): Int {
        return list.size

    }

    override fun onBindViewHolder(holder: UserViewHolder, position: Int) {
        holder.textView.text = list[position].name
        holder.textView2.text = list[position].number
        holder.cardView.setOnLongClickListener(object : View.OnLongClickListener {
            override fun onLongClick(v: View?): Boolean {
                var task: Task<Void> = removeUser(list[position].userId.toString())
                task.addOnCompleteListener { task ->
                    if (task.isSuccessful) {
                        context.showToast("Deleted!!")
                    } else {
                        log(task.exception?.message)
                    }
                }
                return true
            }

        })

    }

    inner class UserViewHolder(itemView: View) : ViewHolder(itemView) {
        var cardView = itemView.cardview
        var textView = itemView.textView_name
        var textView2 = itemView.textView_number


    }
}